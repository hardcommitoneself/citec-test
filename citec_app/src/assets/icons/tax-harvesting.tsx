// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

export const TaxHarvesting = () => {
   return (
      <svg
         width="20"
         height="20"
         viewBox="0 0 20 20"
         fill="currentColor"
         xmlns="http://www.w3.org/2000/svg"
      >
         <path d="M18.772 8.156L17.404 4.051C17.2056 3.45348 16.8238 2.9337 16.3131 2.56551C15.8024 2.19731 15.1886 1.99944 14.559 2H12V0H8V2H5.441C4.81139 1.99944 4.19761 2.19731 3.68688 2.56551C3.17615 2.9337 2.79445 3.45348 2.596 4.051L1.228 8.156C0.864613 8.30848 0.554277 8.56476 0.335845 8.89276C0.117412 9.22077 0.000589089 9.60592 0 10V15C0 15.753 0.423 16.402 1.039 16.743C1.026 16.809 1 16.869 1 16.938V19C1 19.2652 1.10536 19.5196 1.29289 19.7071C1.48043 19.8946 1.73478 20 2 20H3C3.26522 20 3.51957 19.8946 3.70711 19.7071C3.89464 19.5196 4 19.2652 4 19V17H16V19C16 19.2652 16.1054 19.5196 16.2929 19.7071C16.4804 19.8946 16.7348 20 17 20H18C18.2652 20 18.5196 19.8946 18.7071 19.7071C18.8946 19.5196 19 19.2652 19 19V16.938C19 16.869 18.974 16.808 18.961 16.743C19.2744 16.5721 19.5362 16.3202 19.7189 16.0136C19.9017 15.707 19.9988 15.3569 20 15V10C20 9.171 19.492 8.459 18.772 8.156ZM2 15V10H18L18.002 15H2ZM5.441 4H14.558C14.989 4 15.371 4.274 15.507 4.684L16.613 8H3.387L4.492 4.684C4.55835 4.48479 4.68572 4.31153 4.85605 4.18876C5.02638 4.06599 5.23104 3.99995 5.441 4Z" />
      </svg>
   )
}
