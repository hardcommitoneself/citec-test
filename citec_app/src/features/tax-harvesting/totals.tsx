// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { TotalsBox } from 'features/ui/cards'
import { formatToMoney } from 'features/utils/format-to-money'
import { useAppSelector } from 'store/hooks'

export const TotalsTaxHarvesting = () => {
   const taxHarvestingState = useAppSelector((state) => state.taxHarvesting)

   const {
      arrayPositionSelected,
      unrealizedCapitalGains,
      unrealizedCapitalGainsNegative,
   } = taxHarvestingState

   const negativePositionsSelected = arrayPositionSelected
      .map((values) => values.unrealized_capital_gains)
      .filter((values) => values < 0)

   let totalNegativesPositionsSelected = 0

   if (negativePositionsSelected.length > 0) {
      totalNegativesPositionsSelected = negativePositionsSelected.reduce(
         (prev, curr) => prev + curr
      )
   }

   const unrealizedCapitalGainsNumber = formatToMoney(unrealizedCapitalGains)

   const totalHarvestinOpportunity =
      unrealizedCapitalGainsNegative - totalNegativesPositionsSelected

   const harvestingOpportunityNumber = formatToMoney(totalHarvestinOpportunity)

   const lossesHarvestedNumber = formatToMoney(totalNegativesPositionsSelected)

   const cards = [
      {
         img: require('assets/img/bolsa.png'),

         title: ' Unrealized Gains',
         value: unrealizedCapitalGainsNumber,
      },
      {
         img: require('assets/img/espiga.png'),
         title: 'Harvesting Opportunity',
         value: harvestingOpportunityNumber,
      },
      {
         img: require('assets/img/espiga_gray.png'),
         title: 'Losses Harvested',
         value: lossesHarvestedNumber,
      },
   ]

   return (
      <div className="w-full grid grid-cols-3 gap-11 ">
         {cards.map(({ title, value, img }, i) => (
            <TotalsBox key={i} img={img} value={value} text={title} bgWhite />
         ))}
      </div>
   )
}
