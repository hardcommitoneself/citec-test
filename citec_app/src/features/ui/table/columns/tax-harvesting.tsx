// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import { formatToMoney } from 'features/utils/format-to-money'
import { Column } from 'react-table'
import { InputTable } from '../input-table'

export interface Portfolio {
   ticker: string
   unrealized_capital_gains: number
   outstanding_balance: number
}

export const columns: Column<Portfolio>[] = [
   {
      Header: 'Ticker',
      accessor: 'ticker',
      Cell: ({ value }) => <InputTable value={value} />,
   },
   {
      Header: 'Outstanding balance, $',
      accessor: 'outstanding_balance',
      Cell: ({ value }) => <InputTable value={formatToMoney(value, true)} />,
   },
   {
      Header: 'Unrealized capital gains',
      accessor: 'unrealized_capital_gains',
      Cell: ({ value }) => <InputTable value={`(${formatToMoney(value, true)})`} />,
   },
]
