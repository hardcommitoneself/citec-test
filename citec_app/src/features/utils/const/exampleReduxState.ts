// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

export const exampleReduxState = {
   errors: '',
   taxHarvesting: {
      arrayPositionSelected: [
         {
            outstanding_balance: 0,
            ticker: '$CASH',
            unrealized_capital_gains: 500,
            taxAlpha: -200,
         },
      ],
      positionsTable: [
         {
            outstanding_balance: 50000,
            ticker: 'AAPL',
            unrealized_capital_gains: -1000,
            taxAlpha: 400,
         },
         {
            outstanding_balance: 0,
            ticker: '$CASH',
            unrealized_capital_gains: 500,
            taxAlpha: -200,
         },
      ],
      taxAlpha: -400,
      unrealizedCapitalGains: -500,
      unrealizedCapitalGainsNegative: -1000,
      unrealizedCapitalGainsSelected: 0,
   },
   portfolio: {
      ticker: ['$CASH', 'AAPL'],
      outstanding_balance: [1000000, 50000],
      unrealized_capital_gains: [0, 11234],
      analysis: {
         fundamentals: {
            total_assets: 2,
            total_positions: 2,
            pe: 24.32,
            pb: 38.96,
            dividend_yield: 0.59,
         },
         risk: {
            sharpe: 0.305683,
            volatility: 0.0147,
            beta: 0.048719,
            alpha: 0.000035,
         },
         returns: {
            returns_1M: -0.0021,
            returns_3M: -0.0037,
            returns_6M: -0.0028,
            CAGR: -0.1058,
         },
         capital_gains: {
            outstanding_balance: 1050000,
            unrealized_cap_losses: 0,
            unrealized_cap_gains: 11234,
            realized_cap_losses: 0,
            realized_cap_gains: 0,
            taxes_cost: 0,
            taxes_rate: 0.4,
         },
         sectors: {
            sector: ['Cash', 'Information Technology'],
            percentage: [0.952381, 0.047619],
            percentage_benchmark: [0.3333333333333333, 0.6666666666666666],
         },
         factors: {
            factor: ['style', 'size'],
            percentage: [0.502991, 0.52381],
         },
         segment_matrix: {
            style: ['Val', 'Val'],
            size: ['Large', 'Mid'],
            weight: [0, 0],
         },
         top_holdings: {
            ticker: ['$CASH', 'AAPL'],
            weight: [0.952381, 0.047619],
            outstanding_balance: [1000000, 50000],
            unrealized_capital_gains: [0, 11234],
         },
         concentration: {
            ticker: ['$CASH', 'AAPL'],
            concentration: [95.238095, 100],
         },
         var: {
            quantile: [2, 4],
            value: [0.000072, 0.001334],
         },
         risk_returns: {
            ticker: ['portfolio', 'benchmark'],
            returns: [0.004722, 0.119508],
            volatility: [0.008561, 0.205196],
            sharpe: [0.551568, 0.582407],
         },
      },
   },
   universe: {
      factors_filter: [
         {
            factor: 'style',
            ranges: [0, 100],
         },
         {
            factor: 'size',
            ranges: [0, 100],
         },
         {
            factor: 'yield',
            ranges: [0, 100],
         },
      ],
      sectors_filter: [
         'Industrials',
         'Financials',
         'Health Care',
         'Consumer Staples',
         'Communication Services',
         'Information Technology',
         'Consumer Discretionary',
         'Materials',
         'Cash',
         'Real Estate',
         'Energy',
         'Utilities',
      ],
      to_exclude: [],
      to_include: [],
      universe_base: 'SPDR S&P 500',
   },
   rebalance: {
      ticker: ['$CASH', 'AAPL'],
      hold: [0, 1],
      objective: {
         name: 'Minimize tracking error',
         key: 'min_tracking_error',
      },
      benchmark: 'SPY',
      min_weight: 0.000001,
      number_stocks: [10, 200],
   },
   portfolio_optimized: {
      portfolio_id: null,
      portfolio: 'optimized_portfolio',
      benchmark: 'SPY',
      creation_date: null,
      ticker: ['AAPL', 'MSFT'],
      outstanding_balance: [72506.52, 63477.37],
      unrealized_capital_gains: [0, 0],
      optimization_status: 'Optimization terminated successfully',
   },
   trades: [
      {
         ticker: 'MRK',
         operation: 'SELL',
         volume: 275161.76,
         realized_capital_gains: 0,
         unrealized_capital_gains: 0,
         stocks_number: 3053,
         expected_volume: 275193,
         delta_volume: 31,
         price: 90.1388,
         columnName: {
            url_logo:
               'https://static.finnhub.io/logo/2eedcaa6-80ec-11ea-853f-00000000092a.png',
            name: 'Merck & Co Inc',
         },
         gain_loss: [0, 0],
      },
      {
         ticker: 'BMY',
         operation: 'SELL',
         volume: 152166.26,
         realized_capital_gains: 0,
         unrealized_capital_gains: 0,
         stocks_number: 2072,
         expected_volume: 152146,
         delta_volume: -19,
         price: 73.43,
         columnName: {
            url_logo:
               'https://static.finnhub.io/logo/8b628318-80eb-11ea-93a2-00000000092a.png',
            name: 'Bristol-Myers Squibb Co',
         },
         gain_loss: [0, 0],
      },
      {
         ticker: 'WMT',
         operation: 'SELL',
         volume: 83898.27,
         realized_capital_gains: 0,
         unrealized_capital_gains: 0,
         stocks_number: 679,
         expected_volume: 83845,
         delta_volume: -53,
         price: 123.4832,
         columnName: {
            url_logo:
               'https://static.finnhub.io/logo/52057505dd05a752d364065fcddf7772a700b31c47c39e9dcb41e393f6693d52.png',
            name: 'Walmart Inc',
         },
         gain_loss: [0, 0],
      },
   ],
   workflow: {
      type: 'load',
   },
   static_data: {
      universe: {
         universes: [
            {
               name: 'INVESCO QQQ TRUST SERIES 1',
               description: 'INVESCO QQQ TRUST SERIES 1',
            },
            {
               name: 'ISHARES MSCI USA ESG SELECT',
               description: 'ISHARES MSCI USA ESG SELECT',
            },
         ],
         sectors: [
            'Cash',
            'Communication Services',
            'Consumer Discretionary',
            'Consumer Staples',
         ],
         factors: ['style', 'size', 'yield', 'quality'],
      },
      rebalance: {
         objectives: [
            {
               key: 'min_tracking_error',
               name: 'Minimize tracking error',
            },
            {
               key: 'min_risk_parity',
               name: 'Minimize risk parity',
            },
            { key: 'min_volatility', name: 'Minimize volatility' },
            {
               key: 'max_diversification_ratio',
               name: 'Maximize diversification ratio',
            },
         ],
         benchmarks: [
            {
               key: 'sp_500',
               description: 'S&P 500 universe',
               ticker: 'SPY',
            },
            {
               key: 'nasdaq_100',
               description: 'NASDAQ 100',
               ticker: 'QQQ',
            },
         ],
      },
   },
   brokers: {
      brokerConnected: '',
      balance: 0,
      accountType: [],
      a1c_t: '',
   },
   ideas: [
      {
         portfolio_id: 16,
         portfolio: 'my_first_portfolio_sadasd',
         benchmark: 'SPY',
         creation_date: '2022-04-06 16:10:19',
         ticker: ['MSFT', 'AAPL'],
         outstanding_balance: [200000, 100000],
         unrealized_capital_gains: [0, 35700],
         optimization_status: null,
      },
      {
         portfolio_id: 18,
         portfolio: 'testing',
         benchmark: 'SPY',
         creation_date: '2022-04-06 16:10:19',
         ticker: ['MSFT', 'AAPL'],
         outstanding_balance: [200000, 100000],
         unrealized_capital_gains: [0, 35700],
         optimization_status: null,
      },
      {
         portfolio_id: 19,
         portfolio: 'Jueves',
         benchmark: 'SPY',
         creation_date: '2022-06-23 10:38:00',
         ticker: [
            'CMCSA',
            'REGN',
            'VTR',
            'RTX',
            'NEE',
            'AAPL',
            'MMM',
            'TMO',
            'BLK',
            'MSFT',
            'QCOM',
            'NVDA',
            'CSX',
            'ADBE',
            'XOM',
            'AVGO',
            'BAC',
            'TSLA',
            'SHW',
            'TXN',
            'NFLX',
            'GOOGL',
            'C',
            'MMC',
            'BKNG',
            'COST',
            'NKE',
            'DIS',
            'EL',
            'CSCO',
            'BRK.B',
            'META',
            'AMZN',
            'DG',
            'CCI',
            'ORCL',
            'USB',
            'TGT',
            'PG',
            'ICE',
            'GOOG',
            'AMD',
            'MCD',
            'CVS',
            'HD',
            'CME',
            'MU',
            'UNH',
            'ATVI',
            'V',
            'EOG',
            'GS',
            'HUM',
            'COP',
         ],
         outstanding_balance: [
            80352.37, 74072.31, 59343.2, 54736.44, 41924.05, 41510.68, 35441.43,
            27349.74, 26308.48, 26136.99, 25646.73, 25479.44, 24726.53, 24075.27,
            23856.18, 23763.38, 23735.12, 23092.02, 22995.95, 21888.81, 21775.5,
            21090.31, 16734.94, 16192.87, 15720.33, 14020.04, 14018.61, 13225.5,
            12996.74, 11693.52, 11315.64, 10798.53, 10583.72, 10309.54, 10245.54,
            9611.94, 8747.21, 8723.18, 6339.86, 5866.02, 5426.94, 4912.02, 4857.23,
            3915.03, 3395.13, 3370.41, 3322.34, 3184.94, 3001.78, 1983.75, 1858.2,
            1789.12, 1309.98, 1228.49,
         ],
         unrealized_capital_gains: [
            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0, 0, 0,
         ],
         optimization_status: null,
      },
      {
         portfolio_id: 20,
         portfolio: 'Min volatility',
         benchmark: 'SPY',
         creation_date: '2022-06-23 11:25:41',
         ticker: [
            'MRK',
            'BMY',
            'WMT',
            'CMCSA',
            'REGN',
            'VTR',
            'RTX',
            'NEE',
            'MMM',
            'CSX',
            'XOM',
            'SHW',
            'TXN',
            'C',
            'BRK.B',
            'BA',
            'BSX',
            'CB',
            'CI',
            'CL',
            'CME',
            'COP',
            'COST',
            'CRM',
            'ZTS',
            'ADBE',
            'AMZN',
            'AVGO',
         ],
         outstanding_balance: [
            275161.76, 152166.26, 83898.27, 80352.37, 74072.31, 59343.2, 54736.44,
            41924.05, 35441.43, 24726.53, 23856.18, 22995.95, 21888.81, 16734.94,
            11315.64, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49,
            2116.49, 2116.49, 2116.49, 119.48, 96.79, 4.71,
         ],
         unrealized_capital_gains: [
            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0,
         ],
         optimization_status: null,
      },
      {
         portfolio_id: 21,
         portfolio: 'min2',
         benchmark: 'SPY',
         creation_date: '2022-06-23 11:27:32',
         ticker: [
            'MRK',
            'BMY',
            'WMT',
            'CMCSA',
            'REGN',
            'VTR',
            'RTX',
            'NEE',
            'MMM',
            'CSX',
            'XOM',
            'SHW',
            'TXN',
            'C',
            'BRK.B',
            'BA',
            'BSX',
            'CB',
            'CI',
            'CL',
            'CME',
            'COP',
            'COST',
            'CRM',
            'ZTS',
            'ADBE',
            'AMZN',
            'AVGO',
         ],
         outstanding_balance: [
            275161.76, 152166.26, 83898.27, 80352.37, 74072.31, 59343.2, 54736.44,
            41924.05, 35441.43, 24726.53, 23856.18, 22995.95, 21888.81, 16734.94,
            11315.64, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49,
            2116.49, 2116.49, 2116.49, 119.48, 96.79, 4.71,
         ],
         unrealized_capital_gains: [
            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0,
         ],
         optimization_status: null,
      },
   ],
}
