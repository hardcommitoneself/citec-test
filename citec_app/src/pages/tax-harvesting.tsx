// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

/* Definition of terms
From portfolio in Redux arrive 2 terms per positioin
- Outstanding_balance: Balance of that position
- Unrealized_capital_gains: Amount earned if we sell the position now
- Tax rate: 40%
- Tax Alpha: Unrealized_capital_gains * 0.4

Init state of table:
   Order:
      - Unrealized_capital_gains ASCending
   Selected cells:
      - Negative Unrealized_capital_gains

Sketch: https://api.media.atlassian.com/file/957bdc83-5b81-49a1-8516-23823fad63bb/binary?token=eyJhbGciOiJIUzI1NiJ9.eyJpc3MiOiI1YTU4MjVkMC1kMGEwLTRhNGMtOWEwZi1kMGRlMzVhZTVhOWQiLCJhY2Nlc3MiOnsidXJuOmZpbGVzdG9yZTpmaWxlOjk1N2JkYzgzLTViODEtNDlhMS04NTE2LTIzODIzZmFkNjNiYiI6WyJyZWFkIl19LCJleHAiOjE2NTA2OTU2ODQsIm5iZiI6MTY1MDYxMjcwNH0.v-PiP2Q8V8ncPu0CBOXgmx8bv7Vl0um8J-S3BhrpNAs&client=5a5825d0-d0a0-4a4c-9a0f-d0de35ae5a9d&name=skecth_manage_positions.png
*/

import { PageContainer } from 'features/ui/page-container'
import { Button } from '@mui/material'
import { AppLink } from 'routes/AppLink'
import { TableTaxHarvesting, TotalsTaxHarvesting } from 'features/tax-harvesting'

export const TaxHarvesting = () => {
   return (
      <PageContainer titlePage="Tax loss harvesting">
         <TotalsTaxHarvesting />

         <div className="bg-white my-5 p-4 rounded-lg">
            <span className="text-[20px] leading-[30px] text-[#0BACA9] font-[500]">
               Select the positions you want to sell
            </span>

            <TableTaxHarvesting />
         </div>

         <div className="buttons_wrapper">
            <AppLink to="/investment">
               <Button color="secondary">Back</Button>
            </AppLink>
            <AppLink to="/criteria">
               <Button>Next</Button>
            </AppLink>
         </div>
      </PageContainer>
   )
}
