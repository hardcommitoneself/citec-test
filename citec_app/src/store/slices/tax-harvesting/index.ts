// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import { createSlice } from '@reduxjs/toolkit'
import { TaxHarvestingState } from 'features/utils/interfaces/redux/tax-harvesting'

const initialState: TaxHarvestingState = {
   arrayPositionSelected: [],
   unrealizedCapitalGains: 0,
   unrealizedCapitalGainsNegative: 0,
}

export const taxHarvestingSlice = createSlice({
   name: 'tax-harvesting',
   initialState,
   reducers: {
      setPositionSelected: (state, action) => {
         state.arrayPositionSelected = action.payload.arrayPositionSelected
      },
      setInitialPositions: (state, action) => {
         state.unrealizedCapitalGains = action.payload.unrealizedCapitalGains
         state.unrealizedCapitalGainsNegative =
            action.payload.unrealizedCapitalGainsNegative
      },

      resetTaxHarvesting: () => initialState,
   },
})

export const { resetTaxHarvesting, setInitialPositions, setPositionSelected } =
   taxHarvestingSlice.actions

export default taxHarvestingSlice.reducer
