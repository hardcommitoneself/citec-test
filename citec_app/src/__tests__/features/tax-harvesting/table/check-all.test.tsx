// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { Checkbox } from '@mui/material'
import { mount } from 'enzyme'
import { CheckAll } from 'features/tax-harvesting'
import { exampleReduxState } from 'features/utils/const/exampleReduxState'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

describe('test at <CheckAll />', () => {
   const mockStore = configureStore()

   const store = mockStore(exampleReduxState)
   const addOrRemoveAllPositions = jest.fn()

   const wrapper = mount(
      <Provider store={store}>
         <CheckAll addOrRemoveAllPositions={addOrRemoveAllPositions} />
      </Provider>
   )

   test('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot()
   })

   test('should render checkbox mui', () => {
      const checkBox = wrapper.find(Checkbox) as any

      expect(checkBox).toHaveLength(1)
   })

   test('should calls to functions', () => {
      const checkBox = wrapper.find(Checkbox) as any

      checkBox.prop('onChange')({ target: { checked: true } })
      expect(addOrRemoveAllPositions).toBeCalled()

      checkBox.prop('onChange')({ target: { checked: false } })
      expect(addOrRemoveAllPositions).toBeCalled()
   })
})
