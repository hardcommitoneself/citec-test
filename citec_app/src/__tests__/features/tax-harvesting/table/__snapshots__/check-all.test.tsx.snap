// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test at <CheckAll /> should match snapshot 1`] = `ReactWrapper {}`;
